app/controllers/ingredients_controller.rb

```diff
@@ -0,0 +1,46 @@
+class IngredientsController < ApplicationController
+    def new
+    end
+
+    def create
+        @ingredient = Ingredient.new(ingredient_params)
+        if @ingredient.valid?
```

Comment:
You don't have any validations inside ingredient right now. Why should we validate right now? It's going to always say that the model is valid, because we don't have validations :)
Instead you should do something like that:
` if @ingredient.save
   redirect ...
`